doit/dependency.py

```diff
@@ -693,6 +693,12 @@ def get_status(self, task, tasks_dict, get_log=False):
 class UptodateCalculator(object):
     """Base class for 'uptodate' that need access to all tasks
     """
+    
```

Comment:
Something like this would probably do the trick. I'm not sure of your exact requirements though so I'll leave it for now.
```
def __getstate__(self):
    if self.unsafe_pickle:
       return self.__dict__
    else:
        return dict_with_attrs_removed
```